### ISSUE TITLE
[MU4 Issue] macOS: shortcut Cmd+F for "Find" doesn't work

### ISSUE THREAD

**Issue by cbjeukendrup** [musescore/MuseScore#14184]
**To Reproduce**
1. Press Cmd+F

Expected: "Find" panel opens

Actual:
- if there is a selection on the score, then the "Insert measures" popup appears, and the log output says that MuseScore thinks you pressed Ctrl+Ins
- if there is no selection, nothing happens. 

**Platform information**
 - OS: happens on macOS; on Linux it works fine (as reported by Marc); probably macOS specific

**Additional context**
Activating the "Find" panel using the "Edit" menu in the menu bar works fine. So it's purely a shortcuts mapping issue.

**Comment by MarcSabatella** [comment]
In Linux at least, the search box appears.  It just doesn’t actually work - entering a measure number doesn’t take you there.  It select the measure but the view doesn’t change.

**Comment by cbjeukendrup** [comment]
@MarcSabatella Yes, when I open the search box via the Edit menu, I get the same behaviour as you describe. But I think that is covered by #10096.

**Comment by Tantacrul** [comment]
Since this was working not too long ago, it's a bad regression but I'm guessing easy to fix too.

**Comment by cbjeukendrup** [comment]
It looks like this is not as trivial as it seems. It turns out that `MacOSShortcutsInstanceModel::doActivate ` is mapping Ctrl+F to Ctrl+Ins. So I think this is rather something for @Eism, who authored `MacOSShortcutsInstanceModel`.

**Comment by Eism** [comment]
@cbjeukendrup 
I can't reproduce on the nighty build

**Comment by cbjeukendrup** [comment]
@Eism That's strange... I can certainly reproduce still, both on nightlies and my own builds. Deleting my shortcuts.xml doesn't help. Can it be a macOS 13 issue?

**Comment by cbjeukendrup** [comment]
Yes, really strange... I can still reproduce. I deleted my shortcuts.xml once again and I'm 100% sure I'm on the latest master, and it still happens for me.

**Comment by Tantacrul** [comment]
I'm moving this to a P3 since it seems hard to reproduce. This should be the last thing we work on in this release.

**Comment by Tantacrul** [comment]
It happened to me today. (MacOS 12.1)

I then cleared the **command+F** shortcut and re-added it again and then it worked. _Really_ odd.

**Comment by Tantacrul** [comment]
So this is conclusively fixed right? I think we just forgot to close this issue.

**Comment by cbjeukendrup** [comment]
I haven't seen such conclusion anywhere. Anyway, I just retested the latest master and it's still broken for me.

**Comment by MarcSabatella** [comment]
What was fixed is the find command itself - it works now and didn’t until a week ago.  What’s not fixed is the default shortcut on macOS specifically.

**Comment by Tantacrul** [comment]
Got it thanks

**Comment by Eism** [comment]
At some point, I also reproduced the bug. I started debugging and accidentally overwritten the old Cmd+F with the new Cmd+F, like Martin did. Now I can't reproduce.

We need to find the reason why the bug starts to reproduce.

**Comment by cbjeukendrup** [comment]
@Eism I guess you have already tried it, but does the bug come back when you delete your custom shortcuts file? (`~/Library/Application Support/MuseScore/MuseScore4Development/shortcuts.xml`)

Also, strange enough, Martin's "workaround" of clearing and re-adding doesn't seem to work for me. MuseScore keeps interpreting Cmd+F as Cmd+Ins.

In fact, it interprets the `F` key as `Ins`, so this is not limited to the "Find" action. When I am in Note Input mode, and I press F, instead of the note F being entered, a measure is inserted.

**Comment by Eism** [comment]
Yes, after deleting `~/Library/Application Support/MuseScore/MuseScore4Development/shortcuts.xml` the bug started to reproduce again.

And yes, the algorithm for matching possible keys for shortcuts worked incorrectly. Ins interprets as key F. Because of this, the find(or add note F) shortcut was not registered.

Rewriting Cmd+F helped because after writing data to the file and reading, the find shortcut is at the top of the list of shortcuts than other shortcuts with the Ins key - reverse situation, shortcuts with Ins did not work. 

As for the solution, I've added handling for Ins - the Ins key will always be mapped to the Ins key. I did it this way because QCocoaIntegration::possibleKeys for some reason crashes when trying to get keys for NSInsertFunctionKey and we cannot use the algorithm as for other keys.

I hope that Ins is the only such problematic key.
